### PR TITLE
M2-LZ-33

### PR DESCRIPTION
**Pull request by agsta**
Ankieta do makiety

### CODE REVIEW

**Review by jakubrog — CHANGES_REQUESTED** [review]
fix your build and everything will be great

File: src/main/java/lz/modelTesting/configuration/MongoDBConfig.java [hunk 1]
```diff
@@ -12,13 +12,14 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.data.mongodb.repository.config.EnableMongoRepositories;
 
-import java.util.Optional;
+import java.util.*;
```
Comment (mimagiera) [inline src/main/java/lz/modelTesting/configuration/MongoDBConfig.java]: Not a problem but be aware of it:
https://stackoverflow.com/questions/147454/why-using-a-wild-card-with-a-java-import-statement-bad

File: src/main/java/lz/modelTesting/controllers/SurveysController.java [hunk 2]
```diff
@@ -42,6 +45,18 @@ public void addSurvey(@RequestBody String postPayload) {
...
     }
 
+    @PostMapping(value = "/addMockupSurvey", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> addMockupSurvey(@RequestBody String postPayload) {
+        JSONObject jsonObject = new JSONObject(postPayload);
+        String name = jsonObject.getString(NAME);
+        String mockupId = jsonObject.getString(MOCKUP_ID);
+        String body = "{\"questions\":" + jsonObject.getJSONArray(BODY).toString() + '}';
+        Survey survey = new Survey(name, body);
+        survey.setMockupId(new JSONObject(postPayload).getString(MOCKUP_ID));
```
Comment (mimagiera) [inline src/main/java/lz/modelTesting/controllers/SurveysController.java]: * Maybe better create another constructor of survey with mockupId?
* you already have variable `mockupId`

Comment (mimagiera) [reply src/main/java/lz/modelTesting/controllers/SurveysController.java]: + there already exists method `getSurveyFromJson`

File: front/src/components/AddMockupSurvey.js [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+        this.state = {
+            serverUrl: "http://localhost:8080",
+            mockupId: this.props.match.params.id,
+            mockup: "",
+            surveyName: "",
+            questions: [],
+            iframeWidth: '125%',
+            iframeHeight: window.innerHeight,
+            scale: 'scale(0.80)',
+            isFull: false,
```
Comment (Eternalynx) [inline front/src/components/AddMockupSurvey.js]: Probably would have changed name isFull to isFullscreen. isFull might mean different things. isFullscreen is more precise

File: front/src/components/MockupSurvey.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        this.state = {
+            serverUrl: "http://localhost:8080",
+            mockupId: "",
+            mockup: "",
+            surveyName: "",
+            questions: [],
+            iframeWidth: '125%',
+            iframeHeight: window.innerHeight,
+            scale: 'scale(0.8)',
+            isFull: false,
```
Comment (Eternalynx) [inline front/src/components/MockupSurvey.js]: Again, isFull probably should be changed to isFullscreen

**Review by Eternalynx — CHANGES_REQUESTED** [review]
Bardzo dobry kod. Nie widzę większych problemów, jedynie kosmetykę w paru miejscach.

**Review by jakubrog — APPROVED** [review]

**Review by Eternalynx — APPROVED** [review]

**Pull request merged**